sim/deathknight/deathknight.go

```diff
@@ -0,0 +1,286 @@
+		SpellSchool: core.SpellSchoolFrost,
+
+		ResourceType: stats.RunicPower,
+		BaseCost:     baseCost,
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				Cost: baseCost,
+				GCD:  core.GCDDefault,
+			},
+			CD: core.Cooldown{
+				Timer:    deathKnight.NewTimer(),
+				Duration: 6.0 * time.Second,
+			},
+		},
+
+		ApplyEffects: core.ApplyEffectFuncDirectDamage(core.SpellEffect{
+			ProcMask:             core.ProcMaskSpellDamage,
+			BonusSpellCritRating: 0 + float64(deathKnight.Talents.DarkConviction),
+			DamageMultiplier:     1 * (1 + 0.05*float64(deathKnight.Talents.ImprovedIcyTouch)),
+			ThreatMultiplier:     7.0,
+
+			BaseDamage:     core.BaseDamageConfigMagic(227, 245, 0),
+			OutcomeApplier: deathKnight.OutcomeFuncMagicHitAndCritAPScaled(deathKnight.DefaultSpellCritMultiplier(), 0.1),
+
+			OnSpellHitDealt: func(sim *core.Simulation, spell *core.Spell, spellEffect *core.SpellEffect) {
+				if spellEffect.Landed() {
+					deathKnight.SpendFrostRune(sim, spell.FrostRuneMetrics())
+
+					// TODO: Generate runic power
+				}
+			},
+		}),
+	})
+}
+
+func NewDeathKnight(character core.Character, options proto.Player) *DeathKnight {
+	deathKnightOptions := options.GetDeathKnight()
+
+	deathKnight := &DeathKnight{
+		Character: character,
+		Talents:   *deathKnightOptions.Talents,
+		Options:   *deathKnightOptions.Options,
+		Rotation:  *deathKnightOptions.Rotation,
+	}
+
+	maxRunicPower := 100.0
+	if deathKnight.Talents.RunicPowerMastery == 1 {
+		maxRunicPower = 115.0
+	} else if deathKnight.Talents.RunicPowerMastery == 2 {
+		maxRunicPower = 130.0
+	}
+	deathKnight.EnableRunicPowerBar(
+		maxRunicPower,
+		func(sim *core.Simulation) {},
+	)
+
+	deathKnight.EnableRuneBars(
+		func(sim *core.Simulation) {},
+		func(sim *core.Simulation) {},
+		func(sim *core.Simulation) {},
+		func(sim *core.Simulation) {},
+	)
+
+	deathKnight.EnableAutoAttacks(deathKnight, core.AutoAttackOptions{
+		MainHand:       deathKnight.WeaponFromMainHand(deathKnight.DefaultMeleeCritMultiplier()),
+		OffHand:        deathKnight.WeaponFromOffHand(deathKnight.DefaultMeleeCritMultiplier()),
+		AutoSwingMelee: true,
+	})
+
+	deathKnight.AddStatDependency(stats.StatDependency{
+		SourceStat:   stats.Agility,
+		ModifiedStat: stats.MeleeCrit,
+		Modifier: func(agility float64, meleecrit float64) float64 {
+			return meleecrit + (agility/62.5)*core.CritRatingPerCritChance
+		},
+	})
+	deathKnight.AddStatDependency(stats.StatDependency{
+		SourceStat:   stats.Agility,
+		ModifiedStat: stats.Dodge,
+		Modifier: func(agility float64, dodge float64) float64 {
+			return dodge + (agility/84.74576271)*core.DodgeRatingPerDodgeChance
+		},
+	})
+	deathKnight.AddStatDependency(stats.StatDependency{
+		SourceStat:   stats.Strength,
+		ModifiedStat: stats.AttackPower,
+		Modifier: func(strength float64, attackPower float64) float64 {
+			return attackPower + strength*2
+		},
+	})
+
+	return deathKnight
+}
+
+func RegisterDeathKnight() {
```

Comment:
Put this at the top of the file if you don't mind, to be consistent with the other classes.

Reply:
Done